### PR TITLE
[clustering] Fixed strip_clustering when using kernel_regularizer

### PR DESCRIPTION
**Pull request by SaoirseARM**
Added kernel_regularizer to clustering API
Fixed issue where the original weights were being maintained within the graph after clustering & stripping.
Added unit tests for keras kernel_regularizer

### CODE REVIEW

**Review by wwwind — APPROVED** [review]

**Comment by SaoirseARM** [comment]
This PR is no longer needed as [[Clustering] Average Gradient Aggregation](https://github.com/tensorflow/model-optimization/pull/667) fixes this.
